File: app/src/main/java/ga/chrom_web/player/multiplayer/Manager.java
package ga.chrom_web.player.multiplayer;


import javax.inject.Inject;

import io.socket.client.Socket;

public abstract class Manager {

    public static final String EVENT_CONNECTED = "connected";
    public static final String EVENT_DISCONNECT = "disc";
    public static final String EVENT_JOIN = "join";


    public static final String EVENT_PLAY = "play";
    public static final String EVENT_PAUSE = "pause";
    public static final String EVENT_REWIND = "rewind";
    public static final String EVENT_VIDEO_TIME = "video_time";
    public static final String EVENT_LIGHT = "light";
    public static final String EVENT_LOAD = "light";

    public static final String EVENT_MESSAGE = "message";

    @Inject
    Socket socket;

    public Manager() {
        App.getComponent().inject(this);
        subscribeOnEvents();
    }

    abstract void subscribeOnEvents();

    protected void printEmit(String eventName) {
        System.out.println("EMIT " + eventName);
    }
}

File: app/src/main/java/ga/chrom_web/player/multiplayer/Manager.kt
package ga.chrom_web.player.multiplayer


import javax.inject.Inject

import io.socket.client.Socket

abstract class Manager {

    companion object {

        const val EVENT_CONNECTED = "connected"
        const val EVENT_DISCONNECT = "disc"
        const val EVENT_JOIN = "join"

        const val EVENT_PLAY = "play"
        const val EVENT_PAUSE = "pause"
        const val EVENT_REWIND = "rewind"
        const val EVENT_VIDEO_TIME = "video_time"
        const val EVENT_LIGHT = "light"
        const val EVENT_LOAD = "load"

        const val EVENT_MESSAGE = "message"
    }

    @Inject
    lateinit var socket: Socket


    init {
        App.getComponent().inject(this)
        subscribeOnEvents()
    }

    protected abstract fun subscribeOnEvents()

    protected fun printEmit(eventName: String) {
        Utils.debugLog("EMIT " + eventName)
    }
}